packages/web3/src/address/address.test.ts

```diff
@@ -77,134 +150,237 @@ describe('address', function () {
       expect(
         groupOfAddress('2jW1n2icPtc55Cdm8TF9FjGH681cWthsaZW3gaUFekFZepJoeyY3ZbY7y5SCtAjyCjLL24c4L2Vnfv3KDdAypCddfAY')
-      ).toBe(0),
+      ).toBe(0)
       expect(
         groupOfAddress('2jXboVD9p66wrAHkPHx2AQocAzYXUWeppmRT3PuVT3ccxX9u8puTnwLeQ2VbTd4sNkgSEgk1cLbyVGLFshGweJCk1Mr')
-      ).toBe(1),
+      ).toBe(1)
       expect(
         groupOfAddress('2je1yvQHpg8bKCDmvr1koELSNbty5DHrHYRkXomiRNvP5VcsZTK3WisBco2sCtCULM2YbxRxPd7QwhdP2hz9PEQwB1S')
       ).toBe(2),
-      expect(
-        groupOfAddress('2jWukVCejM4Zifz9LvMG4dfR6SEecHLX8VqbswhGwnu61d28B861UhLu3ZmTHu4N14m1kk9rbxreBYzcxta1WPawKzG')
-      ).toBe(3),
-      expect(groupOfAddress('eBrjfQNeyUCuxE4zpbfMZcbS3PuvbMJDQBCyk4HRHtX4')).toBe(0),
-      expect(groupOfAddress('euWxyF55nGTxavL6mgGeMrFdvSRzHor8AmhgPXm8Lm9D')).toBe(1),
-      expect(groupOfAddress('n2pYTzmA27tkp7UNFPhMJpjz3jr5vgessxqJ7kwomBMF')).toBe(2),
-      expect(groupOfAddress('tLf6hDfrUugmxZhKxGoZMpAUBt3NcZ2hrTspTCmZ6JdQ')).toBe(3),
-      expect(groupOfAddress('yya86C6UemCeLs5Ztwjcf2Mp2Kkt4mwzzRpBiG6qQ9kj')).toBe(0),
-      expect(groupOfAddress('yya86C6UemCeLs5Ztwjcf2Mp2Kkt4mwzzRpBiG6qQ9kk')).toBe(1),
-      expect(groupOfAddress('yya86C6UemCeLs5Ztwjcf2Mp2Kkt4mwzzRpBiG6qQ9km')).toBe(2),
-      expect(groupOfAddress('yya86C6UemCeLs5Ztwjcf2Mp2Kkt4mwzzRpBiG6qQ9kn')).toBe(3)
-  })
-
-  it('should calculate the group of lockup script', () => {
-    const bytes0 = new Uint8Array(randomBytes(32))
-    const bytes1 = new Uint8Array(randomBytes(32))
-    const bytes2 = new Uint8Array(randomBytes(32))
-
-    const p2pkh: LockupScript = { kind: 'P2PKH', value: new Uint8Array(bytes0) }
-    const p2pkhAddress = bs58.encode(new Uint8Array([0x00, ...bytes0]))
-    expect(groupOfAddress(p2pkhAddress)).toBe(groupOfLockupScript(p2pkh))
-
-    const p2mpkh: LockupScript = { kind: 'P2MPKH', value: { publicKeyHashes: [bytes0, bytes1, bytes2], m: 2 } }
-    const p2mpkhAddress = bs58.encode(new Uint8Array([0x01, 0x03, ...bytes0, ...bytes1, ...bytes2, 0x02]))
-    expect(groupOfAddress(p2mpkhAddress)).toBe(groupOfLockupScript(p2mpkh))
-
-    const p2sh: LockupScript = { kind: 'P2SH', value: bytes0 }
-    const p2shAddress = bs58.encode(new Uint8Array([0x02, ...bytes0]))
-    expect(groupOfAddress(p2shAddress)).toBe(groupOfLockupScript(p2sh))
-
-    const p2c: LockupScript = { kind: 'P2C', value: bytes0 }
-    const p2cAddress = bs58.encode(new Uint8Array([0x03, ...bytes0]))
-    expect(groupOfAddress(p2cAddress)).toBe(groupOfLockupScript(p2c))
-  })
-
-  it('should extract token id from addresses', () => {
-    expect(binToHex(tokenIdFromAddress('wCTeteGBeSEC54GpkS8jWBzYiYNTBUuTW3WzxGd9yExT'))).toBe(
-      '25469eb0d0d0a55deea832924547b7b166c70a3554fe321e81886d3c18f19d64'
-    ),
+        expect(
+          groupOfAddress('2jWukVCejM4Zifz9LvMG4dfR6SEecHLX8VqbswhGwnu61d28B861UhLu3ZmTHu4N14m1kk9rbxreBYzcxta1WPawKzG')
+        ).toBe(3),
+        expect(groupOfAddress('eBrjfQNeyUCuxE4zpbfMZcbS3PuvbMJDQBCyk4HRHtX4')).toBe(0),
```

Comment:
The following seem to be P2SH and P2C addresses, they should belong to the tests below